### PR TITLE
Test whether a stream stop can deadlock

### PR DESCRIPTION
**Pull request by philburk**
Try to reproduce the bug in #2059.

### CODE REVIEW

File: src/common/AudioStreamBuilder.cpp [hunk 1]
```diff
@@ -94,14 +94,19 @@ Result AudioStreamBuilder::openStream(AudioStream **streamPP) {
 }
 
 Result AudioStreamBuilder::openStreamInternal(AudioStream **streamPP) {
+    static bool versionLogged = false;
```
Comment (robertwu1) [inline src/common/AudioStreamBuilder.cpp:97]: Do we need this? I'm not too big of a fan of static variables like this and openStream() shouldn't realistically be called that often. Also, this is a LOGI and not something like LOGE

File: src/opensles/AudioStreamOpenSLES.cpp [hunk 2]
```diff
@@ -66,9 +66,6 @@ SLuint32 AudioStreamOpenSLES::getDefaultByteOrder() {
 }
 
 Result AudioStreamOpenSLES::open() {
-
-    LOGI("AudioStreamOpenSLES::open() chans=%d, rate=%d", mChannelCount, mSampleRate);
```
Comment (robertwu1) [inline src/opensles/AudioStreamOpenSLES.cpp:70]: I really having a log where the parameters are logged somewhere. If this is not logged anywhere else, I prefer leaving this here.

File: src/common/AudioStreamBuilder.cpp (hunk 1 above)
Comment (philburk) [reply src/common/AudioStreamBuilder.cpp:97]: When I run the unit tests it tries to trigger a deadlock by opening and closing rapidly many times. These version logs spew into the test output and can be very spammy. Maybe I could add a compiler flag that disables this log. Then we could just disable it for the tests. That seems cleaner.  Something like:

    #ifndef OBOE_DISABLE_VERSION_LOG
        LOGI(blah blah);
    #endif

File: src/opensles/AudioStreamOpenSLES.cpp (hunk 2 above)
Comment (philburk) [reply src/opensles/AudioStreamOpenSLES.cpp:70]: This only logs 2 out of a dozen parameters. Maybe we should log more.

This spews into the logs when I am testing.  Maybe:

    #ifndef OBOE_SUPPRESS_LOG_SPAM

File: src/common/AudioStreamBuilder.cpp (hunk 1 above)
Comment (philburk) [reply src/common/AudioStreamBuilder.cpp]: Done

**Review by robertwu1 — APPROVED** [review]

**Pull request merged**